runtime/src/serde_snapshot.rs

```diff
@@ -0,0 +1,614 @@
+    }
+}
+
+// Serializable version of AccountStorageEntry for snapshot format V1_2_0
+#[derive(Clone, Copy, Debug, Default, Eq, PartialEq, Serialize, Deserialize)]
+pub struct SerializableAccountStorageEntryV1_2_0 {
+    id: AppendVecId,
+    accounts_current_len: usize,
+}
+
+impl From<&AccountStorageEntry> for SerializableAccountStorageEntryV1_2_0 {
+    fn from(rhs: &AccountStorageEntry) -> Self {
+        Self {
+            id: rhs.id,
+            accounts_current_len: rhs.accounts.len(),
+        }
+    }
+}
+
+impl Into<AccountStorageEntry> for SerializableAccountStorageEntryV1_2_0 {
+    fn into(self) -> AccountStorageEntry {
+        AccountStorageEntry::new_empty_map(self.id, self.accounts_current_len)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use {
+        super::*,
+        crate::{
+            accounts::{create_test_accounts, Accounts},
+            accounts_db::{get_temp_accounts_paths, tests::copy_append_vecs},
+        },
+        rand::{thread_rng, Rng},
+        solana_sdk::{account::Account, pubkey::Pubkey},
+        std::io::{BufReader, Cursor},
+        tempfile::TempDir,
+    };
+
+    fn check_accounts(accounts: &Accounts, pubkeys: &[Pubkey], num: usize) {
+        for _ in 1..num {
+            let idx = thread_rng().gen_range(0, num - 1);
+            let ancestors = vec![(0, 0)].into_iter().collect();
+            let account = accounts.load_slow(&ancestors, &pubkeys[idx]);
+            let account1 = Some((
+                Account::new((idx + 1) as u64, 0, &Account::default().owner),
+                0,
+            ));
+            assert_eq!(account, account1);
+        }
+    }
+
+    #[test]
+    fn test_accounts_serialize() {
+        solana_logger::setup();
+        let (_accounts_dir, paths) = get_temp_accounts_paths(4).unwrap();
+        let accounts = Accounts::new(paths);
+
+        let mut pubkeys: Vec<Pubkey> = vec![];
+        create_test_accounts(&accounts, &mut pubkeys, 100, 0);
+        check_accounts(&accounts, &pubkeys, 100);
+        accounts.add_root(0);
+
+        let mut writer = Cursor::new(vec![]);
+        accountsdb_to_stream(
+            &mut writer,
+            &*accounts.accounts_db,
+            0,
+            &accounts.accounts_db.get_snapshot_storages(0),
+        )
+        .unwrap();
+
+        let copied_accounts = TempDir::new().unwrap();
+
+        // Simulate obtaining a copy of the AppendVecs from a tarball
+        copy_append_vecs(&accounts.accounts_db, copied_accounts.path()).unwrap();
+
+        let buf = writer.into_inner();
+        let mut reader = BufReader::new(&buf[..]);
+        let (_accounts_dir, daccounts_paths) = get_temp_accounts_paths(2).unwrap();
+        let daccounts = Accounts::new_empty(
+            accountsdb_from_stream(&mut reader, &daccounts_paths, copied_accounts.path()).unwrap(),
+        );
+        check_accounts(&daccounts, &pubkeys, 100);
+        assert_eq!(accounts.bank_hash_at(0), daccounts.bank_hash_at(0));
+    }
```

Comment:
how about adding a really simple test of SerdeContextV1_1_0 version of `reconstruct_accounts_db_via_serialization` to test the old snapshot version code path?

Reply:
What I'm a bit worried is that I think currently the default (SerdeContextV1_2_0) is tested implicitly by other various tests.

Reply:
Serialisation tests cases from accounts_db and bank have been moved to serde_snapshot, and run on both the OLDER and NEWER serialisation styles.